discretize/base/base_mesh.py

```diff
@@ -18,13 +20,31 @@ class BaseMesh(properties.HasProperties, InterfaceMixins):
     """
 
     _REGISTRY = {}
+    _aliases = {
+        'nC': 'n_cells',
+        'nN': 'n_nodes',
+        'nEx': 'n_edges_x',
+        'nEy': 'n_edges_y',
+        'nEz': 'n_edges_z',
```

Comment:
is it less inituitive what [0],[1],[2], are though? Are there cases when it wouldn't be x,y,z? Do I need to know that [0]=x etc?

Reply:
Good point Craig. The main reason and I guess why we talked about it was because the meshes have SO MANY attributes, so it is a way to reduce the attribute-jungle.

But yes, everything is up for discussion. However, as we decided to do it this way for `hx`, `hy`, and `hz` etc it would be consistent. Of course, that very decision can also be discussed ;-)

I think it is clear, it is simply first dimension, second dimension, third dimension. They might not be aligned with your x, y, z.

Reply:
A note about what the 1st, 2nd, 3rd dimensions are in various situations (like cartesian or cylindrical axis) would be useful.  It's always frustrating when you have to guess what dimension the first slice corresponds to.

Reply:
I would not ant to make `n_edges` a tuple quantity, It should inherently be a single integer that describes the total number of `edges` on a mesh, looking forward for unstructured meshes essentially. I do no like the fact that `nEx`, `nEy`, and `nEz` all live on `BaseMesh` currently. 

The meaning of `n_edges_x`(`nEx`) is the number of edges with an `x` (or first dimension) tangent direction.

Regards to notes about what the dimensions correspond to, I feel like that should inherently be left to whoever is using discretize. Normally we think of our `TensorMesh`s as the dimensions are (x, y, z)=(+East, +North, Up). However it is in fact valid fo ANY right handed coordinate system, so if you wanted to treat the dimensions as (x, y, z) = (+North, +East, Down), the returned matrices are still completely valid (Or even if you wanted to oddly rotate the coordinate systems).

Reply:
Fair enough. I remember we discussed it in the meeting when you first brought up the google docs. I thought it is worth thinking about it once again,  But sounds reasonable.